pkg/asset-manager-utils/contracts/aave/SinglePoolAaveATokenAssetManager.sol

```diff
@@ -0,0 +1,119 @@
+
+import "./ILendingPoolAddressesProvider.sol";
+import "./ILendingPool.sol";
+import "./IAaveIncentivesController.sol";
+
+import "../SinglePoolAssetManager.sol";
+import "@balancer-labs/v2-distributors/contracts/interfaces/IMultiRewards.sol";
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+// solhint-disable no-empty-blocks
+// solhint-disable var-name-mixedcase
+// solhint-disable private-vars-leading-underscore
+contract SinglePoolAaveATokenAssetManager is SinglePoolAssetManager {
+    uint16 private constant REFERRAL_CODE = 0;
+
+    IAaveIncentivesController public immutable aaveIncentives;
+    ILendingPool public immutable lendingPool;
+    IERC20 public immutable aToken;
+    IERC20 public immutable stkAave;
+
+    /// @notice staking contract for pool which owns this asset manager
+    IMultiRewards public distributor;
+
+    constructor(
+        IVault _vault,
+        address _token,
+        address _lendingPool,
+        address _aToken,
+        address _aaveIncentives,
+        address _stkAave
+    ) SinglePoolAssetManager(_vault, bytes32(0), _token) {
+        // TODO: pull these from Aave addresses provider
+        lendingPool = ILendingPool(_lendingPool);
+        aToken = IERC20(_aToken);
+
+        aaveIncentives = IAaveIncentivesController(_aaveIncentives);
+        stkAave = IERC20(_stkAave);
+
+        IERC20(_token).approve(_lendingPool, type(uint256).max);
+    }
+
+    /**
+     * @dev Should be called in same transaction as deployment through a factory contract
+     * @param pId - the id of the pool
+     * @param rewardsContract - the address of the rewards contract (to distribut stkAAVE)
+     */
+    function initialise(bytes32 pId, address rewardsContract) public {
+        require(poolId == bytes32(0), "Already initialised");
+        poolId = pId;
+        distributor = IMultiRewards(rewardsContract);
+        IERC20(stkAave).approve(rewardsContract, type(uint256).max);
+    }
+
+    /**
+     * @dev To be used in the case of a staking contract migration
+     */
+    function updateStakingContract(address rewardsContract) external {
+        require(msg.sender == address(distributor), "Must be called by current staking contract");
+        IERC20(stkAave).approve(address(distributor), 0);
+
+        distributor = IMultiRewards(rewardsContract);
+        IERC20(stkAave).approve(rewardsContract, type(uint256).max);
+    }
+
+    /**
+     * @dev Deposits capital into Aave
+     * @param amount - the amount of tokens being deposited
+     * @param aum - the current assets under management of this asset manager
+     * @return the number of shares to mint for the pool
+     */
+    function _invest(uint256 amount, uint256 aum) internal override returns (uint256) {
+        lendingPool.deposit(address(token), amount, address(this), REFERRAL_CODE);
+        if (aum == 0) {
+            return amount;
+        }
+        return amount / aum;
```

Comment:
I don't understand this division, can you expand on it? The documentation here talks about Pool shares, but SinglePoolAM has no shares, and the docs in `SinglePoolAM` say this should return 'the number of deposited tokens', which seems to be just `amount`.